solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -99,42 +96,59 @@ contract InterchainAccountRouter is Router, IInterchainAccountRouter {
      * @param _sender The parent account address on the origin domain.
      * @return The address of the interchain account.
      */
-    function getInterchainAccount(uint32 _origin, address _sender)
+    function getInterchainAccount(uint32 _origin, bytes32 _sender)
         public
         view
-        returns (address)
+        returns (address payable)
     {
         return _getInterchainAccount(_salt(_origin, _sender));
     }
 
+    function getInterchainAccount(uint32 _origin, address _sender)
+        external
+        view
+        returns (address payable)
+    {
+        return getInterchainAccount(_origin, _sender.addressToBytes32());
+    }
+
     /**
      * @notice Returns and deploys (if not already) the interchain account for a given `origin`/`sender` pair.
      * @param _origin The origin domain of the interchain account.
      * @param _sender The parent account address on the origin domain.
      * @return The address of the interchain account.
      */
-    function getDeployedInterchainAccount(uint32 _origin, address _sender)
+    function getDeployedInterchainAccount(uint32 _origin, bytes32 _sender)
         public
         returns (OwnableMulticall)
     {
         bytes32 salt = _salt(_origin, _sender);
-        address interchainAccount = _getInterchainAccount(salt);
+        address payable interchainAccount = _getInterchainAccount(salt);
```

Comment:
Use `getInterchainAccount(_origin, _sender)`?

Reply:
salt is used outside the call